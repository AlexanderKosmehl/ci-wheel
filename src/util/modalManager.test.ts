import { describe, expect, it } from 'vitest';
import {
  initModalManager, openArchiveModal, openImportModal, openSpinResultModal,
} from './modalManager';

describe('modalManager', () => {
  const modalContainer = document.createElement<'div'>('div');
  modalContainer.id = 'modal-container';
  document.body.appendChild(modalContainer);

  initModalManager();

  it('can open the spinResult modal', () => {
    openSpinResultModal('Test');
    expect(modalContainer).toMatchSnapshot();
  });

  it('can open the import modal', () => {
    openImportModal();
    expect(modalContainer).toMatchSnapshot();
  });

  it('can open the archive modal', () => {
    openArchiveModal();
    expect(modalContainer).toMatchSnapshot();
  });
});
