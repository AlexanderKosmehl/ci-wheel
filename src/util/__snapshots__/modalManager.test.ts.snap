// Vitest Snapshot v1

exports[`modalManager > can open the archive modal 1`] = `
<div
  id="modal-container"
>
  <div
    class="_container_0360d7"
  >
    <div
      class="_content_0360d7"
    >
      <div
        class="_header_0360d7"
      >
        <h2
          class="_headerText_0360d7"
        >
          Archiv
        </h2>
        <button
          class="_button_7c1f96 _closeButton_0360d7"
          data-test="modalCloseButton"
        >
          <img
            class="_icon_7c1f96"
            src="/src/assets/close-icon.svg"
          />
        </button>
      </div>
      <div
        class="_archiveModalContainer_6def5c"
      >
        <div
          class="_column_6def5c"
        >
          <h3
            class="_header_6def5c"
          >
            Aktuell
          </h3>
          <ul
            class="_listContainer_6def5c"
          />
        </div>
        <div
          class="_column_6def5c"
        >
          <h3
            class="_header_6def5c"
          >
            Alt
          </h3>
          <ul
            class="_listContainer_6def5c"
          />
        </div>
        <div
          class="_removalContainer_6def5c"
        >
          <img
            class="_removalImage_6def5c"
            src="/src/assets/delete-icon.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`modalManager > can open the import modal 1`] = `
<div
  id="modal-container"
>
  <div
    class="_container_0360d7"
  >
    <div
      class="_content_0360d7"
    >
      <div
        class="_header_0360d7"
      >
        <h2
          class="_headerText_0360d7"
        >
          Import
        </h2>
        <button
          class="_button_7c1f96 _closeButton_0360d7"
          data-test="modalCloseButton"
        >
          <img
            class="_icon_7c1f96"
            src="/src/assets/close-icon.svg"
          />
        </button>
      </div>
      <div
        class="_importModalContainer_e2444d"
      >
        <textarea
          class="_importTextarea_e2444d"
          data-test="modalTextarea"
          placeholder="Neue Einträge zeilenweise einfügen..."
        />
        <div
          class="_importButtonWrapper_e2444d"
        >
          <button
            class="_importButton_e2444d"
            data-test="modalImportButton"
          >
            Importieren
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`modalManager > can open the spinResult modal 1`] = `
<div
  id="modal-container"
>
  <div
    class="_container_0360d7"
  >
    <div
      class="_content_0360d7"
    >
      <div
        class="_header_0360d7"
      >
        <h2
          class="_headerText_0360d7"
        >
          Ergebnis:
        </h2>
        <button
          class="_button_7c1f96 _closeButton_0360d7"
          data-test="modalCloseButton"
        >
          <img
            class="_icon_7c1f96"
            src="/src/assets/close-icon.svg"
          />
        </button>
      </div>
      <div
        class="_resultContainer_7299e4"
      >
        <span
          class="_resultLabel_7299e4"
        >
          Test
        </span>
      </div>
    </div>
  </div>
</div>
`;
